java/src/main/java/io/cucumber/java/InvalidMethodSignatureException.java

```diff
@@ -6,14 +6,19 @@
 import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
-final class InvalidMethodSignatureException extends CucumberException {
+final class InvalidMethodSignatureException
+    extends CucumberException {
 
-    private InvalidMethodSignatureException(String message) {
+    private InvalidMethodSignatureException(final String message) {
         super(message);
     }
 
-    static InvalidMethodSignatureExceptionBuilder builder(Method method) {
+    static InvalidMethodSignatureExceptionBuilder builder(final Method method) {
+        if (Objects.isNull(method)) {
+            throw new IllegalArgumentException("Supplied Method can't be null for InvalidMethodSignatureException");
+        }
```

Comment:
The standard java pattern hand for this is `Objects.requireNonNull` with a static import..

 No need to pass in a message as the line should make it obvious which value was null.

Reply:
I guess I should show this as an example:

```java
    public EmbedEvent(Instant timeInstant, TestCase testCase, byte[] data, String mimeType) {
        super(timeInstant, testCase);
        this.data = Objects.requireNonNull(data);
        this.mimeType = Objects.requireNonNull(mimeType);
        this.name = null;
    }
```

The pass through is really use full.

The exception to the no message thing this is when the message involves user input. You can find some examples of this in the `FeaturePath` 

```java
    public static URI parse(String featureIdentifier) {
        requireNonNull(featureIdentifier, "featureIdentifier may not be null");
        if(featureIdentifier.isEmpty()){
            throw new IllegalArgumentException("featureIdentifier may not be empty");
        }
```
Though I must admit it is not consistently applied.

Reply:
will do tomorrow